File: backend/DataCenterAPI/Models/Equipment.cs
namespace DataCenterAPI.Models;

public class Person
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string Type { get; set; } 
    public required string Status { get; set; } 
    public required string Notes { get; set; } 
    public string? Location { get; set; } 
    public DateTime MetSince { get; set; } 
    public string? Connections { get; set; } 
}

File: frontend/src/types/Equipment.ts
export interface Person {
  id: number;
  name: string;
  type: 'colleague' | 'groupmate' | 'family' | 'classmate';
  status: 'best-friend' | 'friend' | 'acquaintance';
  notes: string;
  location?: string;
  metSince: string;
  connections?: string;
}

export interface PeopleStats {
  total: number;
  bestFriend: number;
  friend: number;
  acquaintance: number;
}

export interface FlowNode {
  id: string;
  type: string;
  position: { x: number; y: number };
  data: {
    label: string;
    person: Person;
  };
}

export interface FlowEdge {
  id: string;
  source: string;
  target: string;
  animated?: boolean;
}
